src/hooks/card/useCardSelectModal.ts

```diff
@@ -0,0 +1,32 @@
+import { MouseEvent, useState } from 'react';
+
+export default function useCardSelectModal() {
+  const [isModalOpen, setIsModalOpen] = useState(true);
+  const [selectedCard, setSelectedCard] = useState({
+    name: '',
+    color: '',
+  });
+
+  const 카드사가선택된 = selectedCard.name !== '';
+
+  const handleClickOpenModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const handleClickCloseModal = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleClickCard = (e: MouseEvent<HTMLButtonElement>) => {
+    const { name, color } = e.currentTarget.dataset;
+
+    if (!name || !color) {
+      return;
+    }
+
+    setSelectedCard({ name, color });
+    setIsModalOpen(false);
+  };
```

Comment:
컴포넌트의 React 상태로 color, name를 가지고 있는데 dataset으로 값을 가져올 필요가 있을까요?

```suggestion
  const handleClickCard = ({ name, color }: { name: string, color: string }) => {
    setSelectedCard({ name, color });
    setIsModalOpen(false);
  };
```

Reply:
처음 생각으로는 `view에 의존적이지 않은 코드를 작성하고자 data attribute를 사용하자!` 라고 생각했던 것 같은데 사실 지금과 같은 상황에서는 지금 상황에서는 고려할 필요는 딱히 없는 것 같습니다.

color, name을 그대로 넣어주는게 좋을 것 같네요!